application/comit_node/src/comit_i_routes.rs

```diff
@@ -1,32 +1,101 @@
+use warp::path;
+// Keep `use warp::path;` separate to stop cargo fmt changing it to
+// `warp::path::{self}`
+use crate::settings::ComitNodeSettings;
 use comit_i::Asset;
 use http::Response;
 use mime_guess;
-use std::borrow::Cow;
+use std::{
+    borrow::Cow,
+    net::{IpAddr, Ipv4Addr},
+};
 use warp::{filters::BoxedFilter, path::Tail, Filter, Rejection, Reply};
 
-pub fn create() -> BoxedFilter<(impl Reply,)> {
-    warp::any()
+pub fn create(settings: ComitNodeSettings) -> BoxedFilter<(impl Reply,)> {
+    let settings = warp::any().map(move || settings.clone());
+
+    let comit_node_config = path!("config" / "comitNode.js")
+        .and(warp::get2())
+        .and(warp::query::<GetConfigQueryParams>())
+        .and(warp::path::end())
+        .and(settings)
+        .and_then(serve_comit_node_config);
+
+    let comit_i = warp::any()
         .and(warp::path::tail())
-        .and_then(|path: Tail| serve(path.as_str()))
-        .boxed()
+        .and_then(|path: Tail| serve_comit_i_file(path.as_str()));
+
+    comit_node_config.or(comit_i).boxed()
 }
 
-fn serve(path: &str) -> Result<impl Reply, Rejection> {
+fn serve_comit_i_file(path: &str) -> Result<impl Reply, Rejection> {
```

Comment:
🙃Why the rename?

Reply:
Because now it can serve comit-i files or the comit node conn details config file.